.github/workflows/server-tests.yaml

```diff
@@ -1,10 +1,8 @@
 on:
   pull_request:
-    branches:
-      - main
-      - develop
+    types: [opened, reopened, edited, synchronize]
     paths:
-      - 'server/**'
+      - "server/**"
```

Comment:
Side Quest: Run this on every PR that is opened, reopened, edited, or synchronized. This ensures the action runs on all PR changes.